server/src/test/java/org/elasticsearch/cluster/coordination/CoordinatorTests.java

```diff
@@ -1418,4 +1421,53 @@ public void testDoesNotPerformElectionWhenRestartingFollower() {
         }
     }
 
+    public void testImprovecOnfigurationPerformsVotingConfigExclusionStateCheck() {
```

Comment:
I'd prefer to test the logic in the assertion directly (by making the method static) rather than this quite complicated setup that catches assertions.

Reply:
I tried this and pushed up the commit 0fb29c3e76656cd13b4c8ee6e523ceb2adff97cc to show the changes.

It appears to me though that it didn't cut down the code needed for the setup (for getting the current `ClusterState`, and 2 new invalid `ClusterState` with `VotingConfigExclusion`). In addition, since we are now no longer testing the assertion in `Cooridnator#improveConfiguration`, it is possible to introduce some bugs that may still call the validation method, or somehow bypass it, and doesn't throw assertion error even when the validation method returns false.

What do you think? Or there's other way to simplify this test method even further?

Reply:
Looks good IMO. Not catching `AssertionError` is an improvement; I'm not too worried about failing to call this assertion, that's not an error that we often make.